core/distributed/preconditioner/schwarz.cpp

```diff
@@ -183,6 +241,39 @@ void Schwarz<ValueType, LocalIndexType, GlobalIndexType>::generate(
         this->set_solver(
             gko::share(parameters_.local_solver->generate(local_matrix)));
     }
+
+    gko::remove_complex<ValueType> cweight =
+        gko::detail::real_impl(parameters_.coarse_weight);
+    if (cweight > 0.0 && cweight <= 1.0) {
+        this->local_weight_ = gko::initialize<matrix::Dense<ValueType>>(
+            {ValueType{1} - ValueType{parameters_.coarse_weight}},
+            this->get_executor());
+        this->coarse_weight_ = gko::initialize<matrix::Dense<ValueType>>(
+            {ValueType{parameters_.coarse_weight}}, this->get_executor());
+    } else {
+        this->local_weight_ = gko::initialize<matrix::Dense<ValueType>>(
+            {ValueType{1.0}}, this->get_executor());
+        this->coarse_weight_ = gko::initialize<matrix::Dense<ValueType>>(
+            {ValueType{1.0}}, this->get_executor());
+    }
+
+    if (parameters_.coarse_level && parameters_.coarse_solver) {
+        this->coarse_level_ =
+            share(parameters_.coarse_level->generate(system_matrix));
+        if (as<multigrid::MultigridLevel>(this->coarse_level_)) {
```

Comment:
This can only return false, if `coarse_level_` has dynamic type `MultigridLevel`, but is the nullptr. Can this actually happen? If it doesn't have the correct dynamic type, then `as` will already throw an exception.

Reply:
I assume you mean only return true?

Reply:
No, but the first comma is confusing, so I remove it.